### PR TITLE
feat(management): be able to enable/disable health-check during some periods

### PR DESCRIPTION
**Pull request by gcusnieux**
fix gravitee-io/issues#4043

### CODE REVIEW

File: jackson/src/main/java/io/gravitee/definition/jackson/datatype/services/core/deser/ScheduledServiceDeserializer.java [hunk 1]
```diff
@@ -39,43 +41,61 @@ public ScheduledServiceDeserializer(Class<T> vc) {
...
-
-            final JsonNode unitNode = triggerNode.get("unit");
-            if (unitNode != null) {
-                trigger.setUnit(TimeUnit.valueOf(unitNode.asText().toUpperCase()));
-            } else {
-                throw ctxt.mappingException("[scheduled-service] Unit is required");
-            }
-            service.setTrigger(trigger);
+        final JsonNode scheduleNode = node.get("schedule");
+        String schedule = null;
```
Comment (ytvnr) [inline jackson/src/main/java/io/gravitee/definition/jackson/datatype/services/core/deser/ScheduledServiceDeserializer.java:46]: you can remove this useless initializer
```suggestion
        String schedule;
```

File: jackson/src/main/java/io/gravitee/definition/jackson/datatype/services/core/deser/ScheduledServiceDeserializer.java [hunk 2]
```diff
@@ -39,43 +41,61 @@ public ScheduledServiceDeserializer(Class<T> vc) {
...
+            Long rate = null;
+            TimeUnit unit = null;
+            if (node.has("trigger")) {
+                final JsonNode triggerNode = node.get("trigger");
+                final JsonNode rateNode = triggerNode.get("rate");
+
+                if (rateNode != null) {
+                    rate = rateNode.asLong();
+                } else {
+                    throw ctxt.mappingException("[scheduled-service] Rate is required");
```
Comment (ytvnr) [inline jackson/src/main/java/io/gravitee/definition/jackson/datatype/services/core/deser/ScheduledServiceDeserializer.java:60]: it is deprecated, do you know if we have a better choice ?

File: model/src/main/java/io/gravitee/definition/model/services/dynamicproperty/DynamicPropertyService.java [hunk 3]
```diff
@@ -52,4 +52,5 @@ public void setConfiguration(DynamicPropertyProviderConfiguration configuration)
     public static String getServiceKey() {
         return SERVICE_KEY;
     }
+
```
Comment (ytvnr) [inline model/src/main/java/io/gravitee/definition/model/services/dynamicproperty/DynamicPropertyService.java:55]:
```suggestion
```

**Review by ytvnr — APPROVED** [review]
good to me, little remarks

File: jackson/src/main/java/io/gravitee/definition/jackson/datatype/services/core/deser/ScheduledServiceDeserializer.java [hunk 4]
```diff
@@ -23,6 +23,8 @@
 import io.gravitee.definition.model.services.schedule.Trigger;
```
Comment (ytvnr) [inline jackson/src/main/java/io/gravitee/definition/jackson/datatype/services/core/deser/ScheduledServiceDeserializer.java:23]: It seems it is not used anymore, can you delete the class ?

**Review by ytvnr — CHANGES_REQUESTED** [review]
i think we can delete Trigger class and all references to it

File: jackson/src/main/java/io/gravitee/definition/jackson/datatype/services/core/deser/ScheduledServiceDeserializer.java (hunk 2 above)
Comment (gcusnieux) [reply jackson/src/main/java/io/gravitee/definition/jackson/datatype/services/core/deser/ScheduledServiceDeserializer.java:59]: Perhaps https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/2.9.8/com/fasterxml/jackson/databind/DeserializationContext.html#instantiationException-java.lang.Class-java.lang.String-

Comment (ytvnr) [reply jackson/src/main/java/io/gravitee/definition/jackson/datatype/services/core/deser/ScheduledServiceDeserializer.java:59]: > Note that most of the time this method should NOT be called; instead, handleMissingInstantiator(java.lang.Class<?>, com.fasterxml.jackson.databind.deser.ValueInstantiator, com.fasterxml.jackson.core.JsonParser, java.lang.String, java.lang.Object...) should be called

**Review by ytvnr — APPROVED** [review]
LGTM

**Pull request merged**